code/world.dm

```diff
@@ -211,6 +211,12 @@ var/list/world_api_rate_limit = list()
 	else if (!world.TgsAvailable() && hard_reset)
 		hard_reset = FALSE
 
+	var/msg = "<br><span class='danger'>The server is restarting.</span><br>You should automatically reconnect in a minute or so...<br><hr><br>"
+	world << msg
+	if(SSchat)
+		SSchat.queue(world, msg)
+	sleep(8) // this gives clients time to receive the message
```

Comment:
There must be a better way to do this.

Reply:
You could manually fire the SS? But that's also a bit ew.

Reply:
Also matt says there is a `to_chat_immediate` you should look at

Reply:
Believe me I did try. There are several issues - world.dm is finicky, very few things are available there. I couldn't even get `for(var/client/C in world)` to work properly.

to_chat_immediate was my first attempt actually, but it didn't work for some reason. The whole proc looks fishy to me since the syntax highlighting is broken there (but that might be simple due to the weird regex).

I did think about firing SSChat manually, but it has a pause check (`MC_TICK_CHECK`), so it wouldn't work either.

I agree the sleep is less than ideal. I'm thinking; what if I made a proc in SSChat that "closes" it, giving one last message and stopping receiving new ones, then forcibly `fire`ing until all messages are depleted?

Reply:
Ah I see; the problem is that without the sleep not even to_chat_immediate works. I believe it's an inherent limitation of sending the resources using byond's `output()` proc.

Though what I could do is use `to_chat_immediate` and move it _after_ the MC shutdown, which guarantees no more messages from SSChat, and making the sleep even shorter (since `to_chat_immediate` should need only 1 tick... I hope).